uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/payment/adyen/AdyenDropInServicePresenter.kt

```diff
@@ -10,24 +11,26 @@ import com.karhoo.uisdk.base.BasePresenter
 import com.karhoo.uisdk.util.ANDROID
 import org.json.JSONObject
 
-class AdyenDropInServicePresenter(service: AdyenDropInServiceMVP.Service,
-                                  private val paymentsService: PaymentsService = KarhooApi.paymentsService) :
+class AdyenDropInServicePresenter(context: Context,
+                                  service: AdyenDropInServiceMVP.Service,
+                                  private val paymentsService: PaymentsService = KarhooApi.paymentsService,
+                                  private val repository: AdyenDropInServiceMVP.Repository =
+                                          AdyenDropInServiceRepository(context = context)) :
         BasePresenter<AdyenDropInServiceMVP.Service>(), AdyenDropInServiceMVP.Presenter {
-
     init {
         attachView(service)
     }
 
     override fun getAdyenPayments(paymentComponentData: JSONObject, returnUrl: String) {
-        view?.clearTransactionId()
+        clearTripId()
         val requestString = createPaymentRequestString(paymentComponentData, returnUrl)
         paymentsService.getAdyenPayments(requestString).execute { result ->
             when (result) {
                 is Resource.Success -> {
                     result.data.let { result ->
                         //TODO Find a better way to store / pass through the transaction id
                         val tripId = result.getString(TRIP_ID)
-                        view?.storeTripId(tripId)
+                        storeTripId(tripId)
```

Comment:
Maybe take out the TODO above if we're going to keep using the SharedPreferences for the trip id.